cherrypicks.go

```diff
@@ -135,3 +141,136 @@ Hello :smile_cat: This PR contains changelog entries. Please, verify the need of
 	}
 	return nil
 }
+
+func branchIsABranchInRepository(log *logrus.Entry, repository, queryBranch string) bool {
+	tdir, err := ioutil.TempDir("", "gitcmd")
+	if err != nil {
+		log.Errorf("Failed to create tempdir with error: %s", err)
+		return false
+	}
+	defer os.Remove(tdir)
+	branches, err := git.Command(
+		"ls-remote",
+		"--heads",
+		"git@github.com:mendersoftware/"+repository+".git",
+	).With(&git.State{Dir: tdir}).Run()
+	if err != nil {
+		log.Error(err)
+		return false
+	}
+	for _, branch := range strings.Split(branches, "\n") {
+		if branch == queryBranch {
+			return true
+		}
+	}
+	return false
+
+}
+
+func cherryPickToBranch(log *logrus.Entry, comment *github.IssueCommentEvent, pr *github.PullRequest, conf *config, targetBranch string) error {
+
+	if !branchIsABranchInRepository(log, comment.GetRepo().GetName(), targetBranch) {
+		return fmt.Errorf("%s is not a branch in %s", targetBranch, comment.GetRepo().GetName())
+	}
+
+	prBranchName := fmt.Sprintf("cherry-%s-%s",
+		targetBranch, pr.GetHead().GetLabel())
+	state, err := git.Commands(
+		git.Command("init", "."),
+		git.Command("remote", "add", "mendersoftware",
+			"git@github.com:mendersoftware/"+comment.GetRepo().GetName()+".git"),
+		git.Command("fetch", "mendersoftware"),
+		git.Command("checkout", "mendersoftware/"+targetBranch),
+		git.Command("checkout", "-b", prBranchName),
+	)
+	defer state.Cleanup()
+	if err != nil {
+		return err
+	}
+
+	if _, err = git.Command("cherry-pick", "-x",
+		pr.GetHead().GetRef()+" ^"+pr.GetBase().GetRef()).
+		With(state).Run(); err != nil {
+		if strings.Contains(err.Error(), "conflict") {
+			return errorCherryPickConflict
+		}
+		return err
+	}
+	if _, err = git.Command("push", "mendersoftware", prBranchName+":"+prBranchName).
+		With(state).Run(); err != nil {
+		return err
+	}
+
+	newPR := &github.NewPullRequest{
+		Title: github.String(fmt.Sprintf("[Cherry %s]: %s",
+			targetBranch, comment.GetIssue().GetTitle())),
+		Head: github.String(prBranchName),
+		Base: github.String(targetBranch),
+		Body: github.String(
+			fmt.Sprintf("Cherry pick of PR: %d\nFor you %s :)",
+				pr.GetID(), comment.Sender.GetName())),
+		MaintainerCanModify: github.Bool(true),
+	}
+	client := clientgithub.NewGitHubClient(conf.githubToken)
+	_, err = client.CreatePullRequest(
+		context.Background(),
+		githubOrganization,
+		comment.GetRepo().GetName(),
+		newPR)
+	if err != nil {
+		return fmt.Errorf("Failed to create the PR for: (%s) %v",
+			comment.GetRepo().GetName(), err)
+	}
+	return nil
+}
+
+func cherryPickPR(log *logrus.Entry, comment *github.IssueCommentEvent, pr *github.PullRequest, conf *config, body string, githubClient clientgithub.Client) error {
+	targetBranches := parseCherryTargetBranches(body)
+	if len(targetBranches) == 0 {
+		return fmt.Errorf("No target branches found in the comment body: %s", body)
```

Comment:
Thinking out loud, but this error, as well as "... is not a branch in..." from `cherryPickToBranch` would be nice to be relayed to the PR. We know that right now is quite hard to find the logs, so maybe we could find about a general way of the bot to always report comments.

In any case, let's just keep this in the back of our heads and not complicate (and add risk in) this PR.

Reply:
Yes, good idea!

Let's kick the can down the road for now, but how about the error logs get batched, and then added, with a drop-down menu, like the build versions are? Then the comment is small, and the logs can be easily perused if you so wish?

Reply:
That would look good, yes. As you said, let's think about it later down the road.